### PR TITLE
Simplify AppHost/ElfUtils

### PR DESCRIPTION
**Pull request by am11**
* Delete unused `struct ElfHeader`.
* Dispose `FileStream` object.
* Use readable `u8` string for ELF magic.

### CODE REVIEW

**Review by vitek-karas — APPROVED** [review]

File: src/installer/managed/Microsoft.NET.HostModel/AppHost/ElfUtils.cs [hunk 1]
```diff
@@ -1,51 +1,34 @@
...
+            if (reader.BaseStream.Length < 16) // EI_NIDENT = 16
             {
-                return EI_MAG0 == 0x7f &&
-                       EI_MAG1 == 0x45 &&
-                       EI_MAG2 == 0x4C &&
-                       EI_MAG3 == 0x46;
+                return false;
             }
 
+            byte[] eIdent = reader.ReadBytes(4);
```
Comment (filipnavara) [inline src/installer/managed/Microsoft.NET.HostModel/AppHost/ElfUtils.cs:25]: Is this .NET 7 project? You can remove the `BinaryReader` and just use `.ReadExactly` on the stream.

Comment (am11) [reply src/installer/managed/Microsoft.NET.HostModel/AppHost/ElfUtils.cs:25]: It’s a netstandard2.0 library, so we can’t use new APIs (new language syntax is fine).

**Comment by vitek-karas** [comment]
@VSadov or @elinor-fung could one of you also take a look and merge if it looks good please?

**Review by elinor-fung — APPROVED** [review]
Thanks, @am11!

**Pull request merged**